jakarta-eclipse/src/org/jakarta/jdt/PersistenceMapKeyDiagnosticsCollector.java

```diff
@@ -0,0 +1,164 @@
+						}
+						
+					}
+					
+					/* ======== MapKeyJoinColumn Diagnostic Checks ========= */
+					for (IMethod method: type.getMethods()) {
+						List<IAnnotation> mapKeyJoinCols = new ArrayList<IAnnotation>();
+						for (IAnnotation annotation: method.getAnnotations()) {
+							if (annotation.getElementName().equals(PersistenceConstants.MAPKEYJOINCOLUMN)) {
+								mapKeyJoinCols.add(annotation);
+							}
+						}
+						if (mapKeyJoinCols.size() <= 1) continue;
+						
+						// If we have multiple MapKeyJoinColumn annotations on a single method we must ensure each has a name and referencedColumnName
+						mapKeyJoinCols.forEach((annotation) -> {
+							boolean isNameSpecified = false;
+							boolean isReferencedColumnNameSpecified = false;
+							try {
+								IMemberValuePair[] memberValues =  annotation.getMemberValuePairs();
+								for (IMemberValuePair mv: memberValues) {
+									if (mv.getMemberName().equals(PersistenceConstants.NAME)) {
+										isNameSpecified = true;
+										continue;
+									}
+									
+									if (mv.getMemberName().equals(PersistenceConstants.REFERENCEDCOLUMNNAME)) {
+										isReferencedColumnNameSpecified = true;
+										continue;
+									}
+								}
+								if (!isNameSpecified || !isReferencedColumnNameSpecified) {
+									diagnostics.add(createDiagnostic(method, unit, "A method with multiple @MapKeyJoinColumn Annotations must specify both name and referencedColumnName"));
+								}
+							} catch (JavaModelException e) {
+								Activator.logException("Error while retrieving member values of @MapKeyJoinColumn Annotation", e);
+							}
+						});
+						
+					}
+					
+					for (IField field: type.getFields()) {
+						List<IAnnotation> mapKeyJoinCols = new ArrayList<IAnnotation>();
+						for (IAnnotation annotation: field.getAnnotations()) {
+							if (annotation.getElementName().equals(PersistenceConstants.MAPKEYJOINCOLUMN)) {
+								mapKeyJoinCols.add(annotation);
+							}
+						}
+						if (mapKeyJoinCols.size() <= 1) continue;
+						
+						mapKeyJoinCols.forEach((annotation) -> {
+							boolean isNameSpecified = false;
+							boolean isReferencedColumnNameSpecified = false;
+							try {
+								IMemberValuePair[] memberValues =  annotation.getMemberValuePairs();
+								for (IMemberValuePair mv: memberValues) {
+									if (mv.getMemberName().equals(PersistenceConstants.NAME)) {
+										isNameSpecified = true;
+										continue;
+									}
+									
+									if (mv.getMemberName().equals(PersistenceConstants.REFERENCEDCOLUMNNAME)) {
+										isReferencedColumnNameSpecified = true;
+										continue;
+									}
+								}
+								if (!isNameSpecified || !isReferencedColumnNameSpecified) {
+									diagnostics.add(createDiagnostic(field, unit, "A field with multiple @MapKeyJoinColumn Annotations must specify both name and referencedColumnName"));
```

Comment:
Should be: "A field with multiple `@MapKeyJoinColumn` annotations must specify both the `name` and `referencedColumnName` attributes in the corresponding `@MapKeyJoinColumn` annotations."